yapapi/engine.py

```diff
@@ -762,8 +856,15 @@ async def reject_proposal(reason: str) -> events.ProposalRejected:
             return await reject_proposal("Score too low")
 
         if ignore_draft or not proposal.is_draft:
-            demand_builder = self._demand_builder
-            assert demand_builder is not None
+            assert self._demand_builder is not None
+            demand_builder = deepcopy(self._demand_builder)
+
+            try:
+                demand_builder = await self.engine._strategy.respond_to_provider_offer(
+                    demand_builder, proposal
+                )
+            except ValueError as e:
+                return await reject_proposal(str(e))
```

Comment:
Hmm, if I understand correctly we're sending this way some unknown data in the `reason` field.
E.g. a bug:
```
class MyStrategy(...):
    async def respond_to_provider_offer(...):
         x = getattr(something, my_super_important_private_key)
```
will send the `AttributeError` with contents of `my_super_important_private_key` somewhere, I guess?

Reply:
well, it only catches a `ValueError` ... so I'd argue that you're wrong here... but will wait with the change until we consider this one...

Reply:
Please change this.